### PR TITLE
feat(autofix): Add repo name to root cause code context snippets

### PR DESCRIPTION
**Pull request by roaga**
Adds repository name alongside file path in code snippets. This is to support code hyperlinks on the frontend. Issue #1004 

Sentry PR: https://github.com/getsentry/sentry/pull/75610

### CODE REVIEW

File: src/seer/automation/autofix/components/root_cause/models.py [hunk 1]
```diff
@@ -11,18 +11,21 @@
 
 class SnippetPromptXml(PromptXmlModel, tag="code"):
     file_path: str = attr()
+    repo_name: str = attr()
```
Comment (jennmueng) [inline src/seer/automation/autofix/components/root_cause/models.py:14]: Hmm I think we should let `repo_name` be optional in the cases where there's only one repo, we don't enforce it on the tools in that case

Comment (jennmueng) [reply src/seer/automation/autofix/components/root_cause/models.py:14]: If we're not making it optional, we should make sure we're giving the llm the repo name in the prompts

Comment (roaga) [reply src/seer/automation/autofix/components/root_cause/models.py]: We do need it sent to the frontend no matter what, but I made the field `Optional` for flexibility within Seer

Comment (jennmueng) [reply src/seer/automation/autofix/components/root_cause/models.py]: If there's only one repo then on the frontend we can just make it implied that it's that one whenever it's null here

**Review by jennmueng — APPROVED** [review]

**Pull request merged**